### PR TITLE
perform index deletion in a background job - no more hook required

### PR DESCRIPTION
**Pull request by DeepDiver1975**
@LukasReschke @butonic @icewind1991 please have a look - THX

### CODE REVIEW

File: jobs/deletejob.php [hunk 1]
```diff
@@ -0,0 +1,69 @@
+<?php
+/**
+ * ownCloud - search_lucene
+ *
+ * This file is licensed under the Affero General Public License version 3 or
+ * later. See the COPYING file.
+ *
+ * @author ThomasMüller <deepdiver@owncloud.com>
```
Comment (LukasReschke) [inline jobs/deletejob.php]: I like your new name

File: jobs/deletejob.php [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+ * @copyright Thomas Müller 2015
+ */
+
+namespace OCA\Search_Lucene\Jobs;
+
+use OCA\Search_Lucene\AppInfo\Application;
+use OCA\Search_Lucene\Core\Logger;
+use OC\BackgroundJob\TimedJob;
```
Comment (LukasReschke) [inline jobs/deletejob.php]: That is private API and not even used anymore here?

Comment (DeepDiver1975) [inline jobs/deletejob.php]: we have currently no public base class - not nice but well ... I suspect we need to change these mechanisms anyhow with respect to async processing, queues and so on

File: jobs/deletejob.php [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+		$deletedIds = $mapper->getDeleted();
+		$count = 0;
+		foreach ($deletedIds as $fileId) {
+			$logger->debug( 'deleting status for ('.$fileId.') ' );
+			//delete status
+			$status = new Status($fileId);
+			$mapper->delete($status);
+			//delete from lucene
+			$count += $index->deleteFile($fileId);
+
```
Comment (LukasReschke) [inline jobs/deletejob.php]: Unneeded newline?

**Comment by DeepDiver1975** [comment]
by moving the index delete operation we reduced the actual delete time by another 9 minutes:
https://blackfire.io/profiles/7efb175a-68cf-4e4e-8970-2d7bd6089eff/graph

Quite understandable as there is no delete hook any more :wink:

**Comment by LukasReschke** [comment]
:+1:

**Comment by DeepDiver1975** [comment]
thx - as a result we should very close have a look at our file hooks -> tech dept 8.1

**Comment by butonic** [comment]
nice will start going through all the issues i was @mentioned in in the last week.

**Comment by DeepDiver1975** [comment]
> nice will start going through all the issues i was @mentioned in in the last week.

THX a lot!

**Comment by butonic** [comment]
:+1: Will submit a separate PR for cosmatic changes.

**Pull request merged**